components/EditCard.js

```diff
@@ -0,0 +1,169 @@
+                required
+              />
+              <br />
+              <input
+                type="checkbox"
+                id="condition1"
+                name="conditions"
+                value="Good weather"
+                onChange={handleConditionChange}
+              />
+              <label htmlFor="conditin1">Good weather</label>
+              <br />
+              <input
+                type="checkbox"
+                id="condition2"
+                name="conditions"
+                value="Sunny"
+                onChange={handleConditionChange}
+              />
+              <label htmlFor="condition2">Sunny</label>
+              <br />
+              <input
+                type="checkbox"
+                id="condition3"
+                name="conditions"
+                value="Cloudy"
+                onChange={handleConditionChange}
+              />
+              <label htmlFor="condition3">Cloudy</label>
+              <br />
+              <input
+                type="checkbox"
+                id="condition4"
+                name="conditions"
+                value="Rainy"
+                onChange={handleConditionChange}
+              />
+              <label htmlFor="condition4">Rainy</label>
+              <br />
+              <input
+                type="checkbox"
+                id="condition5"
+                name="conditions"
+                value="Snowy"
+                onChange={handleConditionChange}
+              />
+              <label htmlFor="condition5">Snowy</label>
+              <br />
+              <label htmlFor="date">Date:</label>
+              <br />
+              <input
+                type="date"
+                id="date"
+                name="date"
+                placeholder="Date"
+                required
+              />
+              <br />
+              <label htmlFor="friends">Friends:</label>
+              <br />
+              <div>
+                <input
+                  type="text"
+                  id="friends"
+                  name="friends"
+                  placeholder="Friends"
+                  value={inputFriend}
+                  onChange={handleInputChange}
+                />
+                <button type="button" onClick={handleAddFriend}>
+                  Add
+                </button>
+              </div>
+              <ul>
+                {friends.map((friend, index) => (
+                  <li key={index}>
+                    {friend}
+                    <button
+                      type="button"
+                      onClick={() => handleDeleteFriend(friend)}
+                    >
+                      Delete
+                    </button>
+                  </li>
+                ))}
+              </ul>
+            </div>
+            <div>
+              <button type="submit">
+                {" "}
+                onClick={() => onEditTrip(trip)} Save changes
+              </button>
```

Comment:
if this is a submit button it most likely should not have an onClick event. Also you are calling onEditTrip on the form submit.